src/test/scala/org/broadinstitute/dsde/firecloud/service/TSVFileSupportSpec.scala

```diff
@@ -51,4 +55,77 @@ class TSVFileSupportSpec extends AnyFreeSpec with TSVFileSupport {
     }
   }
 
+  "setAttributesOnEntity" - {
+    "parse an attribute array consisting of all strings" in {
+      val resultingOps = setAttributesOnEntity("some_type", None, MockTSVLoadFiles.entityWithAttributeStringArray.tsvData.head, Seq(("arrays", None)), FlexibleModelSchema)
+      resultingOps.operations.size shouldBe 4 //1 to remove any existing list, 3 to add the list elements
+      resultingOps.entityType shouldBe "some_type"
+
+      val resultingOpNames = resultingOps.operations.map(ops => ops("op"))
+
+      //assert that these operations always remove the existing attribute by this name
+      resultingOpNames.head shouldBe AttributeString("RemoveAttribute")
+
+      //assert that all of the remaining operations are adding a list member
+      resultingOpNames.tail.map { op =>
+        op shouldBe AttributeString("AddListMember")
+      }
+
+      println(resultingOps)
+
+    }
+
+    "parse an attribute array consisting of all numbers" in {
+      val resultingOps = setAttributesOnEntity("some_type", None, MockTSVLoadFiles.entityWithAttributeNumberArray.tsvData.head, Seq(("arrays", None)), FlexibleModelSchema)
+      resultingOps.operations.size shouldBe 4 //1 to remove any existing list, 3 to add the list elements
+      resultingOps.entityType shouldBe "some_type"
+
+      val resultingOpNames = resultingOps.operations.map(ops => ops("op"))
+
+      //assert that these operations always remove the existing attribute by this name
+      resultingOpNames.head shouldBe AttributeString("RemoveAttribute")
+
+      //assert that all of the remaining operations are adding a list member
+      resultingOpNames.tail.map { op =>
+        op shouldBe AttributeString("AddListMember")
+      }
+    }
+
+    "parse an attribute array consisting of all booleans" in {
+      val resultingOps = setAttributesOnEntity("some_type", None, MockTSVLoadFiles.entityWithAttributeBooleanArray.tsvData.head, Seq(("arrays", None)), FlexibleModelSchema)
+      resultingOps.operations.size shouldBe 4 //1 to remove any existing list, 3 to add the list elements
+      resultingOps.entityType shouldBe "some_type"
+
+      val resultingOpNames = resultingOps.operations.map(ops => ops("op"))
+
+      //assert that these operations always remove the existing attribute by this name
+      resultingOpNames.head shouldBe AttributeString("RemoveAttribute")
+
+      //assert that all of the remaining operations are adding a list member
+      resultingOpNames.tail.map { op =>
+        op shouldBe AttributeString("AddListMember")
+      }
+    }
```

Comment:
I'd love to see somewhat deeper validation in these tests. Something like:
```
    def validateListMemberOps(ops: Seq[Map[String, Attribute]], expectedClass: Class[_], expectedAttrName: String = "arrays") = {
      ops.map { op =>
        op.keySet should contain theSameElementsAs List("attributeListName", "newMember", "op")
        op("attributeListName") shouldBe AttributeString(expectedAttrName)
        val element = op("newMember")
        element.getClass shouldBe(expectedClass)
      }
    }
```
and then within each test, you can call:
```
validateListMemberOps(resultingOps.operations.tail, AttributeString("").getClass)
```
(substituting `AttributeString`, `AttributeNumber`, `AttributeBoolean` as appropriate)